okhttp-testing-support/src/main/kotlin/okhttp3/OkHttpClientTestRule.kt

```diff
@@ -175,9 +182,25 @@ class OkHttpClientTestRule : TestRule {
           }
 
           Thread.setDefaultUncaughtExceptionHandler(defaultUncaughtExceptionHandler)
-          ensureAllConnectionsReleased()
-          releaseClient()
-          ensureAllTaskQueuesIdle()
+          try {
```

Comment:
try/catch and throw inside finally is icky; it’s really difficult control flow to follow.

Maybe something like this will work better?

```
val failure: Throwable? = null

try {
 // ...
} catch (t: Throwable) {
  if (failure != null) failure = t;
}

try {
 // cleanup 1
} catch (t: Throwable) {
  if (failure != null) failure = t;
}

try {
 // cleanup 2
} catch (t: Throwable) {
  if (failure != null) failure = t;
}

if (failure != null) throw t
```

Reply:
One other option is to make closeables for the three cleanup tasks (ensureAllConnectionsReleased(), releaseClient(), ensureAllTaskQueuesIdle()) and use `use {}` on each of ’em. That also permits suppressed exceptions!

Reply:
In this case we want the test to fail if our test is causing these to break.  So we don't want to return a client that has active connections.  I'll rework this slightly, but the change as is should hopefully not hide the underlying problem.

I'm pretty sure this previously didn't hide the root cause, so not sure why it's regressed.